ontolearn/refinement_operators.py

```
@@ -562,29 +565,28 @@ def refine_data_some_values_from(self, ce: OWLDataSomeValuesFrom) -> Iterable[OW
 
         datarange = ce.get_filler()
         if isinstance(datarange, OWLDatatypeRestriction) and ce.get_property() in self.dp_splits:
-            new_facet_res: List[OWLFacetRestriction] = []
             splits = self.dp_splits[ce.get_property()]
-            for facet_res in datarange.get_facet_restrictions():
+            if len(splits) > 0 and len(datarange.get_facet_restrictions()) > 0:
+                facet_res = datarange.get_facet_restrictions()[0]
```

In general, the refinement operator will just add DatatypeRestrictions with only one facet restriction. The [DL-Learner implementation](https://github.com/SmartDataAnalytics/DL-Learner/blob/e2b5ba8729a8ee2eff412d524fd6e63c6863cd0c/components-core/src/main/java/org/dllearner/refinementoperators/RhoDRDown.java#L695) loops through the FacetRestrictions and takes the last one, if there are somehow more than one. That should not happen or is there something I’m missing here? I just take the first one, we could also throw an error or do nothing if there is more than one but I think this is fine.